mne/export/_edf.py

```diff
@@ -0,0 +1,239 @@
+    hdl = EDFwriter(fname, file_type, n_channels)
+
+    # set channel data
+    for ichan, ch in enumerate(ch_names):
+        if physical_range == 'auto':
+            # take the channel type minimum and maximum
+            ch_type = ch_types[ichan]
+            pmin, pmax = ch_types_phys_min[ch_type], ch_types_phys_max[ch_type]
+
+        for key, val in [('PhysicalMaximum', pmax),
+                         ('PhysicalMinimum', pmin),
+                         ('DigitalMaximum', digital_max),
+                         ('DigitalMinimum', digital_min),
+                         ('PhysicalDimension', phys_dims),
+                         ('SampleFrequency', channel_sfreq),
+                         ('SignalLabel', ch),
+                         ('PreFilter', filter_str_info)]:
+            _try_to_set_value(hdl, key, val, channel_index=ichan)
+
+    # set patient info
+    subj_info = raw.info.get('subject_info')
+    if subj_info is not None:
+        birthday = subj_info.get('birthday')
+        first_name = subj_info.get('first_name')
+        last_name = subj_info.get('last_name')
+        name = None
+        if first_name is not None:
+            name = first_name
+        if last_name is not None:
+            name += last_name
+        if name is None:
+            name = ''
+
+        hand = subj_info.get('hand')
+        sex = subj_info.get('sex')
+
+        if birthday is not None:
+            if hdl.setPatientBirthDate(birthday[0], birthday[1],
+                                       birthday[2]) != 0:
+                raise RuntimeError(f"Setting Patient Birth Date to {birthday} "
+                                   f"returned an error")
+        for key, val in [('PatientName', name),
+                         ('PatientGender', sex),
+                         ('AdditionalPatientInfo', f'hand={hand}')]:
+            _try_to_set_value(hdl, key, val)
+
+    # set measurement date
+    meas_date = raw.info['meas_date']
+    if meas_date:
+        subsecond = meas_date.microsecond / 100.
+        if hdl.setStartDateTime(year=meas_date.year, month=meas_date.month,
+                                day=meas_date.day, hour=meas_date.hour,
+                                minute=meas_date.minute,
+                                second=meas_date.second,
+                                subsecond=subsecond) != 0:  # noqa
+            raise RuntimeError(f"Setting Start Date Time {meas_date} "
+                               f"returned an error")
+
+    device_info = raw.info.get('device_info')
+    if device_info is not None:
+        device_type = device_info.get('type')
+        _try_to_set_value(hdl, 'Equipment', device_type)
+
+    # set data record duration
+    if data_record_duration is not None:
+        _try_to_set_value(hdl, 'DataRecordDuration', data_record_duration)
+
+    # compute number of seconds to loop over
+    n_secs = n_times / channel_sfreq
+
+    # Write each second (i.e. datarecord) separately.
```

Comment:
here again, we're not guaranteed that 1 datarecord == 1 second

```suggestion
    # Write each datarecord sequentially
```